pkg/controller/cronjob/utils.go

```diff
@@ -63,17 +63,23 @@ func deleteFromActiveList(sj *batchv2alpha1.CronJob, uid types.UID) {
 
 // getParentUIDFromJob extracts UID of job's parent and whether it was found
 func getParentUIDFromJob(j batchv1.Job) (types.UID, bool) {
-	creatorRefJson, found := j.ObjectMeta.Annotations[v1.CreatedByAnnotation]
-	if !found {
-		glog.V(4).Infof("Job with no created-by annotation, name %s namespace %s", j.Name, j.Namespace)
+	creatorRefJson, err := makeCreatedByRefJson(&j)
+
+	if err != nil {
+		glog.V(4).Infof("Failed to create object reference, name %s namespace %s: %v", j.Name, j.Namespace, err)
 		return types.UID(""), false
 	}
+
 	var sr v1.SerializedReference
-	err := json.Unmarshal([]byte(creatorRefJson), &sr)
+	err = json.Unmarshal([]byte(creatorRefJson), &sr)
```

Comment:
yup! i will use `GetControllerOf()` to obtain the deserialized object. thanks for the explanation!